src/main/java/com/ibm/watson/apis/conversation_enhanced/discovery/DiscoveryClient.java

```diff
@@ -0,0 +1,140 @@
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.ibm.watson.apis.conversation_enhanced.discovery;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.ibm.watson.apis.conversation_enhanced.payload.DocumentPayload;
+import com.ibm.watson.apis.conversation_enhanced.utils.Constants;
+import com.ibm.watson.apis.conversation_enhanced.utils.Messages;
+import com.ibm.watson.developer_cloud.discovery.v1.model.query.QueryResponse;
+
+public class DiscoveryClient {
+
+	private Pattern pattern = Pattern.compile("((.+?)</p>){1,2}");
+
+	private static final Logger logger = LogManager.getLogger(DiscoveryClient.class.getName());
+
+	/**
+	 * This method uses the Query object to send the user's query (the
+	 * <code>input</code> param) to the discovery service
+	 * 
+	 * @param input
+	 *            The user's query to be sent to the discovery service
+	 * @return A list of DocumentPayload objects, each representing a single
+	 *         document the retrieve and rank service believes is a possible
+	 *         answer to the user's query
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	public List<DocumentPayload> getDocuments(String input) throws Exception {
+		List<DocumentPayload> documents = new ArrayList<DocumentPayload>();
+		DiscoveryQuery discoveryQuery = new DiscoveryQuery();
+		QueryResponse output = discoveryQuery.query(input);
+		documents = createPayload(input, new Gson().toJson(output.getResults()));
+		return documents;
+	}
+
+	/**
+	 * Helper Method to include highlighting information along with the retrieve
+	 * and rank response so the final payload includes id,title,body,sourceUrl
+	 * as json key value pairs.
+	 * 
+	 * @param input
+	 *            The user's query sent to the discovery service
+	 * @param results
+	 *            The results obtained from a call to the retrieve and rank
+	 *            service with <code>input</code> as the query
+	 * @return A list of DocumentPayload objects, each representing a single
+	 *         document the discovery service believes is a possible answer to
+	 *         the user's query
+	 */
+	private List<DocumentPayload> createPayload(String input, String results) {
+		logger.info(Messages.getString("Service.CREATING_DISCOVERY_PAYLOAD"));
```

Comment:
I don't know how `results` look like, but if it is coming from Discovery, it should be an object not an string.

In the worse case, make "results" to be an object by using Gson(it comes with the java-sdk)

Something like:
```java
MyResultObject resultObject = new Gson().parse(results);
```